events.cpp

```diff
@@ -0,0 +1,267 @@
+	this->factionNum = unit->faction->num;
+	this->unitName = unit->name->Str();
+	this->unitNum = unit->num;
+}
+
+void BattleSide::AssignArmy(Army* army) {
+	this->total = army->count;
+
+	for (int i = 0; i < army->count; i++) {
+		auto soldier = army->soldiers[i];
+        bool lost = soldier->hits == 0;
+
+        if (lost) this->lost++;
+
+        ItemType& item = ItemDefs[soldier->race];
+
+        if (item.flags & ItemType::MANPRODUCE) {
+            this->fmi++;
+            if (lost) this->fmiLost++;
+            continue;
+        }
+        
+        if (item.type & IT_UNDEAD) {
+            this->undead++;
+            if (lost) this->undeadLost++;
+            continue;
+        }
+
+        if (item.type & IT_MONSTER) {
+            this->monsters++;
+            if (lost) this->monstersLost++;
+            continue;
+        }
+
+        if (soldier->unit->type == U_MAGE || soldier->unit->type == U_GUARDMAGE) {
+            this->mages++;
+            if (lost) this->magesLost++;
+            continue;
+        }
+	}
+}
+
+std::string EventLocation::getTerrain() {
+    return TerrainDefs[this->terrainType].name;
+}
+
+void EventLocation::Assign(ARegion* region) {
+	this->x = region->xloc;
+	this->y = region->yloc;
+	this->z = region->zloc;
+	this->terrainType = region->type;
+	this->province = region->name->Str();
+    
+    if (region->town) {
+        this->settlement = region->town->name->Str();
+        this->settlementType = region->town->TownType();
+    }
+}
+
+
+/////-----
+
+
+Events::Events() {
+}
+
+Events::~Events() {
+    for (auto &fact : this->facts) {
+        delete fact;
+    }
+
+    this->facts.clear();
+}
+
+void Events::AddFact(FactBase *fact) {
+    this->facts.push_back(fact);
+}
+
+bool compareEvents(const Event &first, const Event &second) {
+    return first.score < second.score;
+}
+
+std::list<string> wrapText(std::string input, std::size_t width) {
+    std::size_t curpos = 0;
+    std::size_t nextpos = 0;
+
+    std::list<std::string> lines;
+    std::string substr = input.substr(curpos, width + 1);
+
+    while (substr.length() == width + 1 && (nextpos = substr.rfind(' ')) != input.npos) {
+        lines.push_back(input.substr(curpos, nextpos));
+        
+        curpos += nextpos + 1;
+        substr = input.substr(curpos, width + 1);
+    }
+
+    if (curpos != input.length()) {
+        lines.push_back(input.substr(curpos, input.npos));
+    }
+
+    return lines;
+}
+
+std::string makeLine(std::size_t width, bool odd, std::string text)  {
+    std::string line = (odd ? "( " : " )");
+
+    std::size_t left = (width - text.size()) / 2;
+
+    while (line.size() < left) line += ' ';
+    line += text;
+    while (line.size() < (width - 2)) line += ' ';
+
+    line += (odd ? " )" : "(");
+    line += "\n";
+
+    return line;
+}
+
+std::string Events::Write(std::string worldName, std::string month, int year) {
+    std::list<Event> events;
+
+    // // test start
+
+    // auto f = new BattleFact();
```

Comment:
should it be removed?

Reply:
Don't have unit tests here, should be left so that later easier to make testing